website_sale_close/controllers/main.py

```diff
@@ -0,0 +1,127 @@
+# -*- coding: utf-8 -*-
+# Copyright 2019 Coop IT Easy SCRLfs
+#   - Rémy Taymans <remy@coopiteasy.be>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from ast import literal_eval
+
+from openerp import http
+from openerp.addons.website_sale.controllers.main import website_sale
+from openerp.http import request
+
+
+class WebsiteSaleCloseController(website_sale):
+    """
+    Overwrite routes from website_sale controller to close the
+    e-commerce when required by configuration.
+    """
+
+    def closed_ecommerce(self):
+        return request.render(
+            "website_sale_close.closed_ecommerce",
+            {"text_closed_ecommerce": request.website.text_closed_ecommerce},
+        )
+
+    @http.route()
+    def shop(self, *args, **kw):
```

Comment:
Generally, pythonistas write `**kwargs` for keywords arguments rather than `**kw`

Reply:
Ok, 'will change it. :)

Reply:
Done.